cmd/config/config.go

```diff
@@ -0,0 +1,51 @@
+package config
+
+import (
+	"github.com/spf13/viper" // a popular configuration management library for Go.
+)
+
+// DBConfig holds the info needed for connecting to the database
+// It's annotated with mapstructure tags to indicate how configuration keys should be mapped to struct fields.
+type DBConfig struct {
+	Host         string `mapstructure:"HOST"`
+	Port         int    `mapstructure:"PORT"`
+	User         string `mapstructure:"USER"`
+	Password     string `mapstructure:"PASSWORD"`
+	DatabaseName string `mapstructure:"DATABASE_NAME"`
+}
+
+// ServerConfig Holds information about the server (host and port).
+type ServerConfig struct {
+	Host string `mapstructure:"HOST"`
+	Port int    `mapstructure:"PORT"`
+}
+
+// Config The main configuration struct that contains nested fields for database connection information and server information.
+// It's also annotated with mapstructure tags.
+type Config struct {
+	DBConfig   DBConfig     `mapstructure:"DBConfig"`
+	ServerInfo ServerConfig `mapstructure:"ServerConfig"`
+}
+
+var vp *viper.Viper
```

Comment:
If vp will be declared inside the function, it will be recreated every time the function is called. While this isn't a significant issue for performance, it might lead to some overhead in creating and initializing the viper instance repeatedly.